clouds/redshift/libraries/python/lib/clustering/__init__.py

```diff
@@ -10,24 +10,76 @@ def load_geom(geom):
     geom = json.dumps(_geom)
     return loads(geom)
 
+
 def remove_duplicated_coords(arr):
     import numpy as np
+
     unique_rows = []
     for row in arr:
         if not any(np.array_equal(row, unique_row) for unique_row in unique_rows):
             unique_rows.append(row)
     return np.array(unique_rows)
 
+
+def reorder_coords(coords):
+    import numpy as np
+
+    unique_coords = []
+    duplicated_coords = []
+
+    # Split the array into unique and duplicated coordinates
+    count_map = {}
+    for coord in coords:
+        coord_str = tuple(coord)
+        if coord_str not in count_map:
+            count_map[coord_str] = 1
+            unique_coords.append(coord)
+        else:
+            count_map[coord_str] += 1
+            duplicated_coords.append(coord)
+
+    # Convert lists to NumPy arrays for sorting
+    unique_coords = np.array(unique_coords)
+    duplicated_coords = np.array(duplicated_coords)
+
+    # Sort unique coordinates lexicographically if not empty
+    if unique_coords.size > 0:
+        if duplicated_coords.size > 0:
+            # Concatenate unique and duplicated coordinates
+            return np.concatenate((unique_coords, duplicated_coords))
+        else:
+            return unique_coords
+    else:
+        # Sort duplicated coordinates lexicographically if not empty
```

Comment:
@vdelacruzb I would remove this comment as the coordinates are no longer sorted lexicographically

Reply:
Done